src/libstat/backends/redis_backend.c

```diff
@@ -1428,13 +1425,29 @@ rspamd_redis_parse_classifier_opts (struct redis_stat_ctx *backend,
 
 	elt = ucl_object_lookup (obj, "prefix");
 	if (elt == NULL || ucl_object_type (elt) != UCL_STRING) {
+		gchar *redis_object;
+
 		/* Default non-users statistics */
 		if (backend->enable_users || backend->cbref_user != -1) {
-			backend->redis_object = REDIS_DEFAULT_USERS_OBJECT;
+			redis_object = REDIS_DEFAULT_USERS_OBJECT;
 		}
 		else {
-			backend->redis_object = REDIS_DEFAULT_OBJECT;
+			redis_object = REDIS_DEFAULT_OBJECT;
 		}
+
+		/* Prepend classifier name if defined */
+		elt = ucl_object_lookup (obj, "name");
+		if (elt != NULL && ucl_object_type (elt) == UCL_STRING) {
+			const gchar *cl_name = ucl_object_tostring (elt);
+			gchar *temp;
+			temp = g_malloc (strlen (cl_name) + strlen (redis_object) + 2);
+			strcpy (temp, cl_name);
+			strcat (temp, "_");
+			strcat (temp, redis_object);
```

Comment:
This leaks memory and is terribly ugly. Please use mempool and rspamd_snprintf instead.